src/ol/controls/RoutingControl.js

```diff
@@ -0,0 +1,359 @@
+        this.map.removeInteraction(this.modify);
+        unByKey(this.onMapClickKey);
+        return;
+      }
+
+      // Define and add modify interaction
+      this.modify = new Modify({
+        source: this.routingLayer.olLayer.getSource(),
+        pixelTolerance: 4,
+        deleteCondition: (e) => {
+          const feats = e.target.getFeaturesAtPixel(e.pixel, {
+            hitTolerance: 5,
+          });
+          const viaPoint = feats.find(
+            (feat) => feat.getGeometry() instanceof Point && feat.get('index'),
+          );
+          if (click(e) && viaPoint) {
+            this.removeViaPoint(viaPoint.get('index'));
+            return true;
+          }
+          return false;
+        },
+      });
+
+      this.modify.on('modifystart', (e) => {
+        // Create instances for LineString and Point features
+        const route = e.features
+          .getArray()
+          .find((feat) => feat.getGeometry() instanceof LineString);
+
+        const viaPointFeatures = e.features
+          .getArray()
+          .filter((feat) => feat.getGeometry() instanceof Point);
+
+        // Add listeners to viaPoint features to identify which one is being modified
+        this.relocateViaPoint = (evt) => {
+          this.initialRouteDrag.relocateViaPoint = evt.target;
+        };
+
+        viaPointFeatures.forEach((feature) => {
+          feature.on('change', this.relocateViaPoint);
+        });
+
+        // Write object with modify info
+        this.initialRouteDrag = {
+          viaPointFeatures,
+          oldRoute: route && route.clone(),
+          coordinate: e.mapBrowserEvent.coordinate,
+        };
+      });
+
+      this.modify.on('modifyend', (e) => {
+        /* Get the index for new viaPoint */
+        let newViaIndex;
+        const {
+          oldRoute,
+          relocateViaPoint,
+          viaPointFeatures,
+        } = this.initialRouteDrag;
+
+        // Unlisten feature change listeners
+        viaPointFeatures.forEach((feature) => {
+          feature.un('change', this.relocateViaPoint);
+        });
+
+        // If viaPoint is being relocated overwrite the old viaPoint
+        if (relocateViaPoint) {
+          return this.addViaPoint(
+            e.mapBrowserEvent.coordinate,
+            relocateViaPoint.get('index'),
+            1,
+          );
+        }
+
+        // In case there is no route overwrite first coordinate
+        if (!oldRoute) {
+          return this.addViaPoint(e.mapBrowserEvent.coordinate, 0, 1);
+        }
+
+        // Get the closest point from viaPoints to original route
+        const viaPointClosestPoints = this.viaPoints.map((viaPoint) => {
+          const snapPoint = oldRoute.getGeometry().getClosestPoint(viaPoint);
+          return snapPoint;
+        });
+
+        // Calculate the distance from the click point to each segment
+        let distanceToSegment;
+        for (let i = 0; i < viaPointClosestPoints.length - 1; i += 1) {
```

Comment:
you have a forEachSegment function in ol.geom.LineString

Reply:
don't know if it helps :)

Reply:
@oterral  Don't see how this method helps us in this case. We need the segments between the viaPoints, not the segments of the LineString, to be able to identify the new viaPoint index